jupyterhub/proxy.py

```diff
@@ -61,8 +112,8 @@ def add_route(self, routespec, target, data):
         """Add a route to the proxy.
 
         Args:
-            routespec (str): A specification for which this route will be matched.
-                Could be either a url_prefix or a fqdn.
+            routespec (RouteSpec or str): A specification for which this route will be matched.
```

Comment:
I disagree. When approximately all cases are passing a trivial wrapper for a single string whose interpretation is unambiguous and trivial to handle, the API should handle it.